platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBaseDslScriptModelBuilder.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.tooling.builders
+
+import org.gradle.internal.build.BuildState
+import org.gradle.kotlin.dsl.provider.KotlinScriptClassPathProvider
+import org.gradle.kotlin.dsl.support.ImplicitImports
+import org.gradle.tooling.model.kotlin.dsl.KotlinBaseDslScriptModel
+import org.gradle.tooling.provider.model.internal.BuildScopeModelBuilder
+import java.io.File
+import java.io.Serializable
+
+
+internal
+object KotlinBaseDslScriptModelBuilder : BuildScopeModelBuilder {
+    override fun canBuild(modelName: String): Boolean =
+        modelName == "org.gradle.tooling.model.kotlin.dsl.KotlinBaseDslScriptModel"
+
+    override fun create(target: BuildState): KotlinBaseDslScriptModel {
+        val implicitImports = target.getMutableModel().getServices().get(ImplicitImports::class.java)
```

Comment:
💅 We likely have an applicable `serviceOf<T>` utility

Reply:
It seems I can use:
```
 target.getMutableModel().serviceOf<>()
 ```
 
 Thanks!